receiver/googlecloudspannerreceiver/internal/filterfactory/filterbymetricbuilder.go

```diff
@@ -0,0 +1,154 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterfactory
+
+import (
+	"errors"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/filter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/metadata"
+)
+
+type filterByMetricBuilder struct {
+	logger *zap.Logger
+	config *ItemFilterFactoryConfig
+}
+
+func (b filterByMetricBuilder) buildFilterByMetricZeroTotalLimit() map[string]filter.ItemFilter {
+	filterByMetric := make(map[string]filter.ItemFilter)
+	nopItemFilter := filter.NewNopItemCardinalityFilter()
```

Comment:
I need to come back to this, however, if this is doing nothing then is the loop needed?

Reply:
Also, considering this is new code, can I ask that we adopt more go idiomatic variable names (short, concise variables).

In this case It would be:
```suggestion
nopFilter := filter.NewNopItemCardinalityFilter()
```

Reply:
It is done to be consistent with case when we creating Op filter per metric. In case of Nop same pointer is used as value for all metrics(and I do not want to place  lot of if statements in Nop case) - with such approach same handling and filter resolving in both cases. Also Nop just returns same items as it received.

Reply:
Will rename the variable.

Reply:
Done.